code/modules/mob/say.dm

```diff
@@ -168,6 +168,9 @@
 					// No, you cannot speak in xenocommon just because you know the key
 					if(!can_speak_language(LD))
 						return message
+					// you are not allowed to use metalanguage key
+					if(LD == /datum/language/metalanguage && !HAS_TRAIT(src, TRAIT_METALANGUAGE_KEY_ALLOWED))
+						return message
```

Comment:
Should probably move this into the  can_speak_language proc

Reply:
It's a secondary check if you're really allowed to do that so because it shouldn't be visible or controllable at most times.